pkg/safe/routine.go

```diff
@@ -46,19 +48,55 @@ func (p *Pool) Ctx() context.Context {
 	return p.baseCtx
 }
 
+func genID() string {
```

Comment:
shouldn't we use a UUID if this is supposed to be random?

Reply:
I (honestly) don't know. why? what's the problem with the current approach that UUID would solve?

Reply:
the larger number of connections you have, the larger the risk of a collision.

That would result in a random thread being removed no?

Using a UUID would prevent this because it is almost assuredly unique

Reply:
> the larger number of connections you have, the larger the risk of a collision.
> 
> That would result in a random thread being removed no?

in theory yes, in practice no. :-)
I think I've made sure that we never reuse an ID that already exists in the Pool.
But I'll look at UUIDs anyway, I wonder what the overhead is.

> Using a UUID would prevent this because it is almost assuredly unique

Reply:
@dtomcej looking at the implementation here: https://github.com/google/uuid/blob/master/version4.go#L29 , it looks like what we're doing in genID is pretty similar, and at least as strong wrt to collisions (maybe even stronger, since we have 160 bits instead of 122), so I'm not worried about leaving it as is. WDYT?

Reply:
LGTM

Reply:
Sorry for breaking in. I accidentally saw your conversation. What if we use ints instead of strings, as map keys? Then we can use unix timestamp and atomic counter, it will allocate only 64 bit per key (for example, twitter snowflakes unique ids)
Something like this:

```
func (p *Pool) genID() uint64 {
    var id uint64
    epoch := uint64(time.Now().UTC().UnixNano()) / 10000
    id = epoch << (64 - 48)                      // 48 bits for epoch
    id |= uint64(p.counter % (1<<16 - 1))        // 16 bits for counter
    atomic.AddUint64(&p.counter, 1)
    return id
```
Pool's counter can be uint64, for example. It will take much time to overload it.
And we can hold 65536 calls per 0.01 ms.
What do you think?

Reply:
thanks, but unless necessary, I'd rather not make genID more complex than it needs to be. especially now that I've seen that the uuid package does something pretty similar.